src/components/vm/disks/diskAdd.jsx

```diff
@@ -504,11 +505,38 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                // Limit process time to 2 seconds and ram to 100 MB
+                cockpit.spawn(["/bin/bash", "-c", "ulimit -t 2 -v 100000; qemu-img info --output json " + value], { err: "message" })
+                        .then(output => {
+                            try {
+                                const data = JSON.parse(output);
+                                if (data['backing-filename']) {
+                                    this.dialogErrorSet(_("Importing an image with a backing file is unsupported"),
+                                                        cockpit.format(_("Backing file: $0"), data['backing-filename']));
+                                    this.setState({ customDiskVerified: false });
+                                } else {
+                                    this.setState({ customDiskVerified: true, format: data.format, dialogError: "" });
```

Comment:
And then add a separate case for "format" which handles iso files:
```
          case 'format': {
              if (value === "iso")
                  // use onValueChange instead of setState in order to perform subsequent state change logic
                  this.onValueChanged("device", "cdrom");
              this.setState({ format: value });
              break;
          }
```